### PR TITLE
trino request handler fix for exception message length

### PR DESCRIPTION
**Pull request by satish-mittal**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/TrinoBasedRequestHandler.java [hunk 1]
```diff
@@ -130,8 +130,7 @@ public Observable<Row> handleRequest(QueryRequest request, ExecutionContext exec
 
       return executeQuery(sql.getKey(), sql.getValue());
     } catch (Throwable t) {
-      String truncatedMessage = (t.getMessage() == null) ? null : t.getMessage().substring(0, 2048);
-      return Observable.error(new Throwable(truncatedMessage));
+      return Observable.error(new Throwable(truncateMessage(t.getMessage(), 2048)));
```
Comment (kotharironak) [inline query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/TrinoBasedRequestHandler.java:133]: Can we use existing library function?
```
StringUtils.truncate(t.getMessage(), 2048)
```
Also define maxExceptionLength as private static varibale.

Comment (kotharironak) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/TrinoBasedRequestHandler.java:133]: StringUtils.truncate takes care of null check, out of index array, etc.

Comment (kotharironak) [inline query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/TrinoBasedRequestHandler.java:133]:
```suggestion
      return Observable.error(new Throwable(StringUtils.truncate(t.getMessage(), 2048)));
```

Comment (satish-mittal) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/TrinoBasedRequestHandler.java:133]: Wanted to avoid 3p lib dependency

**Review by kotharironak — APPROVED** [review]

**Pull request merged**